react/Wrapper.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { FC } from 'react'
+import useProduct from 'vtex.product-context/useProduct'
+import { withToast } from 'vtex.styleguide'
+
+import AddToCartButton from './AddToCartButton'
+import { mapCatalogItemToCart } from './utils'
+
+interface Props {
+  isOneClickBuy: boolean
+  available: boolean
+  disabled: boolean
+  customToastUrl: string
+  showToast: Function
+}
+
+const Wrapper: FC<Props> = ({
+  isOneClickBuy,
+  available,
+  disabled,
+  customToastUrl,
+  showToast,
+}) => {
+  const productContext: ProductContextState | undefined = useProduct()
+
+  if (!productContext) {
+    throw new Error('useProduct must be used within a ProductContextProvider')
```

Comment:
Who would handle this error?

Reply:
This error would cause the `AddToCart` component to not be rendered. And that's desirable because this should not be used outside of a `ProductContext.Provider`